api/api_admin.go

```diff
@@ -142,19 +142,49 @@ func (a *API) handleGetMCPTools(c *gin.Context) {
 
 	mcpConfig := a.config.MCP()
 
-	// If MCP is not enabled or no servers configured, return empty response
-	if !mcpConfig.Enabled || len(mcpConfig.Servers) == 0 {
+	// If MCP is not enabled, return empty response
+	if !mcpConfig.Enabled {
 		c.JSON(http.StatusOK, MCPToolsResponse{
 			Servers: []MCPServerInfo{},
 		})
 		return
 	}
 
+	// Calculate total capacity (remote servers + embedded server if enabled)
```

Comment:
Probably excessive memory saving. Maybe just len(mcpConfig.Servers)+1 without the check?